torch/optim/optimizerw.py

```diff
@@ -0,0 +1,98 @@
+import torch
+from .optimizer import Optimizer
+from .adam import Adam
+from .sgd import SGD
+
+
+class OptimizerW(Optimizer):
+    r"""Base class for decoupled weight decay optimizers.
+
+    The distinction between weight decay and L2 regularisation is discussed in
+    `Decoupled Weight Decay Regularization`_.
+
+    .. warning::
+        Parameters need to be specified as collections that have a deterministic
+        ordering that is consistent between runs. Examples of objects that don't
+        satisfy those properties are sets and iterators over values of dictionaries.
+
+    Arguments:
+        params (iterable): an iterable of :class:`torch.Tensor` s or
+            :class:`dict` s. Specifies what Tensors should be optimized.
+        defaults: (dict): a dict containing default values of optimization
+            options (used when a parameter group doesn't specify them).
+
+    .. _Decoupled Weight Decay Regularization:
+        https://arxiv.org/abs/1711.05101
+    """
+
+    @torch.no_grad()
+    def step(self, closure=None):
+        """Performs a single optimization step (parameter update).
+
```

Comment:
There is no note field here, unlike `Optimizer.step`. Is that intentional?

Reply:
That is not intentional. Should the note also be added to `get[_sparse]_update` then?

Reply:
yes, let's add it too